libcudacxx/include/cuda/std/__random/seed_seq.h

```diff
@@ -0,0 +1,192 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA_STD___RANDOM_SEED_SEQ_H
+#define _CUDA_STD___RANDOM_SEED_SEQ_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__algorithm/copy.h>
+#include <cuda/std/__algorithm/fill.h>
+#include <cuda/std/__algorithm/max.h>
+#include <cuda/std/cstddef>
+#include <cuda/std/cstdint>
+#include <cuda/std/initializer_list>
+
+#include <cuda/std/__cccl/prologue.h>
+
+_CCCL_BEGIN_NAMESPACE_CUDA_STD
+
+/// @class seed_seq
+///
+/// Generate unbiased seeds by filling the output range [begin, end) with 32-bit
+/// unsigned integer values, based on the (possibly biased) seeds stored in the
+/// internal buffer. If `begin == end`, the function does nothing. Otherwise,
+/// the values are produced by the standard seed sequence mixing algorithm.
+///
+/// If `std::iterator_traits<RandomIt>::value_type` is not an unsigned integer
+/// type, or its width is less than 32 bits, the program is ill-formed.
+/// If `RandomIt` does not meet the requirements of a LegacyRandomAccessIterator
+/// or is not mutable, the behavior is undefined.
+class _CCCL_TYPE_VISIBILITY_DEFAULT seed_seq
+{
+public:
+  using result_type = ::cuda::std::uint_least32_t;
+
+  /// @brief Default-construct an empty seed sequence.
+  ///
+  /// `size()` will return 0.
+  seed_seq() = default;
+
+  /// @brief Construct from an iterator range of seed values.
+  /// @tparam _InputIt Input iterator type yielding values convertible to `result_type`.
+  /// @param __begin Iterator pointing to the first seed value.
+  /// @param __end Iterator one-past-the-last seed value.
+  template <class _InputIt>
+  _CCCL_API _CCCL_CONSTEXPR_CXX20 seed_seq(_InputIt __begin, _InputIt __end)
+  {
+    auto __n = ::cuda::std::distance(__begin, __end);
+    if (__n <= 0)
+    {
+      return;
+    }
+    __data_ = new result_type[__n];
+    __size_ = __n;
```

Comment:
```suggestion
    __size_ = static_cast<size_t>(__n);
```
Let's explicitly convert signed to unsigned